### PR TITLE
🚀 Added way to get specific header of request

### PR DESCRIPTION
**Pull request by Norbiros**
Now you can get header and read it due to #23.
It is nessesary if you want get for example browser that connect to your webpage, or language of request.

**Tested on minecraft server!**

### CODE REVIEW

**Comment by LowScarlet** [comment]
how to use it

**Comment by Norbiros** [comment]
For example like this:
```if header "Accept-Language" of the request starts with "en-US":```

In some files, you can find tags like `@Examples`
```
@Name("Request Header")
@Description({"When connection is made, client send data about it. ",
        "You can access to them using this code!)"})
@Since("1.2.1")
@Examples({"return header \"User-Agent\" of the request",
        "if header \"Accept-Language\" of the request starts with \"en-US\":"})
```

**Comment by ItsTheSky** [comment]
Use `SimplePropertyExpression` instead of `SimpleExpression`.
If you still want to keep the actual code, don't forget to null-check the expression & the parsed value.

**Comment by Norbiros** [comment]
Sorry, but I don't understand. This code is practically copy of request's param, if there we don't use it, we I need to use it here? But yes, I can use it...

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package dev.jensderuiter.websk.skript.expression;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:7]:
```suggestion
import ch.njol.Skript.expressions.base.SimplePropertyExpression;
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+@Name("Request Header")
+@Description({"When connection is made, client send data about it. ",
+        "You can acces to them using this code!)"})
+@Since("1.2.1")
+@Examples({"return header \"User-Agent\" of the request",
+        "if header \"Accept-Language\" of the request starts with \"en-US\":"})
+
+public class ExprRequestHeader extends SimpleExpression<String> {
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:29]:
```suggestion
public class ExprRequestHeader extends SimplePropertyExpression<String> {
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    static {
+        Skript.registerExpression(ExprRequestHeader.class, String.class, ExpressionType.COMBINED, "[the] header %string% of %request%");
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:32]:
```suggestion
        register(ExprRequestHeader.class, String.class, "header", "request");
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    private Expression<String> param;
+    private Expression<Request> request;
+
+    @Override
+    public Class<? extends String> getReturnType() {
+        return String.class;
+    }
+
+    @Override
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:43]:
```suggestion

```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    public boolean isSingle() {
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:44]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        return true;
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:45]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    }
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:46]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    @SuppressWarnings("unchecked")
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:48]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:49]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parser) {
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:50]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        param = (Expression<String>) exprs[0];
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:51]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        request = (Expression<Request>) exprs[1];
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:52]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+        return true;
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:53]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+    }
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:54]:
```suggestion
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public String toString(Event event, boolean debug) {
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:57]:
```suggestion
    protected String getPropertyName() {
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+        return "header " + this.param.toString(event, debug) + " of request";
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:58]:
```suggestion
        return "header";
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:61]:
```suggestion
    @Override
    @Nullable
```

File: src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+    protected String[] get(Event event) {
```
Comment (Norbiros) [inline src/main/java/dev/jensderuiter/websk/skript/expression/ExprRequestHeader.java:62]:
```suggestion
    public String get(Event event) {
```

**Review by Norbiros — COMMENTED** [review]
Something like that @SkyCraft78?

**Review by ItsTheSky — APPROVED** [review]
Yep exactly!
Gonna take a look at the other expression you specified.

**Pull request merged**